packages/components/date-field/src/date-field.ts

```diff
@@ -0,0 +1,290 @@
+   */
+  @property() placeholder?: string;
+
+  /**
+   * Whether the date field is readonly.
+   * @default false
+   */
+  @property({ type: Boolean }) readonly?: boolean;
+
+  /**
+   * Whether the date field is a required field.
+   * @default false
+   */
+  @property({ type: Boolean, reflect: true }) override required?: boolean;
+
+  /**
+   * Whether the component is select only. This means you cannot type in the text field,
+   * but you can still pick a date via de popover.
+   * @default false
+   */
+  @property({ type: Boolean, reflect: true, attribute: 'select-only' }) selectOnly?: boolean;
+
+  /**
+   * Shows the week numbers.
+   * @default false
+   */
+  @property({ type: Boolean, attribute: 'show-week-numbers' }) showWeekNumbers?: boolean;
+
+  /** The selected date in the calendar. */
+  @property({ converter: dateConverter }) override value?: Date;
+
+  /** @internal The wrapper element that is also the popover. */
+  @query('[part="wrapper"]') wrapper?: HTMLSlotElement;
+
+  override connectedCallback(): void {
+    super.connectedCallback();
+
+    if (!this.input) {
+      this.input = this.querySelector<HTMLInputElement>('input[slot="input"]') || document.createElement('input');
+      this.input.autocomplete = 'off';
+      this.input.slot = 'input';
+
+      if (!this.input.parentElement) {
+        this.append(this.input);
+      }
+    }
+
+    this.setFormControlElement(this.input);
+
+    // This is a workaround, because :has is not working in Safari and Firefox with :host element as it works in Chrome
+    const style = document.createElement('style');
+    style.innerHTML = `
+      sl-date-field:has(input:hover):not(:focus-within)::part(text-field) {
+        --_bg-opacity: var(--sl-opacity-light-interactive-plain-hover);
+      }
+    `;
+    this.prepend(style);
+  }
+
+  override willUpdate(changes: PropertyValues<this>): void {
+    super.willUpdate(changes);
+
+    if (changes.has('dateTimeFormat') && changes.has('locale')) {
+      this.#formatter = new Intl.DateTimeFormat(this.locale, this.dateTimeFormat);
+    }
+
+    if (changes.has('value')) {
+      console.log('value', this.value);
+
+      this.input.value = this.value && this.#formatter ? this.#formatter.format(this.value) : '';
+    }
+  }
+
+  override render(): TemplateResult {
+    return html`
+      <sl-text-field
+        @input=${this.#onInput}
+        @keydown=${this.#onKeydown}
+        @sl-blur=${this.#onTextFieldBlur}
+        @sl-change=${this.#onTextFieldChange}
+        @sl-focus=${this.#onTextFieldFocus}
+        @sl-form-control=${this.#onTextFieldFormControl}
+        @sl-update-state=${this.#onTextFieldUpdateState}
+        ?disabled=${this.disabled}
+        ?readonly=${this.readonly || this.selectOnly}
+        part="text-field"
+        placeholder=${ifDefined(this.placeholder)}
+      >
+        <slot name="input" slot="input"></slot>
+        <sl-field-button
+          @click=${this.#onButtonClick}
```

Comment:
The calendar can be opened with keyboard only when using `space`, is it a proper and intuitive behaviour?

<img width="689" alt="image" src="https://github.com/user-attachments/assets/34d0d9b2-cdbf-4e68-917b-6703f42aa798" />

Reply:
And I cannot move to calendar with keyboard to operate in the calendar.